contracts/CarLease.sol

```diff
@@ -269,15 +274,12 @@ contract CarLease {
     function deleteContract(address leasee, bool refundDeposit) internal {
         Contract memory con = contracts[leasee];
 
-        uint excessPayed = 0;
-        
-        if (con.amountPayed > (con.monthlyQuota * getDurationInMonths(con.duration))) {
-            excessPayed = con.amountPayed - (con.monthlyQuota * getDurationInMonths(con.duration));
-        }
-
         if (refundDeposit) {
-            payable(leasee).transfer(3*con.monthlyQuota + excessPayed);
-            transferrableAmount -= excessPayed;
+            uint refundableAmount = 3*con.monthlyQuota;
+            if (con.extended == ContractExtensionStatus.PROPOSED) {
+                refundableAmount += getNewMonthlyQuota(leasee);
```

Comment:
I think this deserves a coment
`// If the contract extension has been proposed but not accepted, we refund the pre-paid 1st month's rent when deleting the contract `